### PR TITLE
PF nested field MaxItemsOne

### PR DESCRIPTION
**Pull request by iwahbe**
Fixes https://github.com/pulumi/pulumi-terraform-bridge/issues/1013

### CODE REVIEW

File: pf/internal/schemashim/block_schema.go [hunk 1]
```diff
@@ -70,10 +77,19 @@ func (s *blockSchema) Elem() interface{} {
...
 	switch tt := t.(type) {
 	case types.MapType:
+		if r, ok := retObj(tt.ElemType); ok {
+			return r
+		}
 		schema = newTypeSchema(tt.ElemType, s.block.NestedAttrs())
 	case types.ListType:
+		if r, ok := retObj(tt.ElemType); ok {
+			return r
+		}
```
Comment (iwahbe) [inline pf/internal/schemashim/block_schema.go:87]: This is the change in behavior that fixed the problem.

File: pkg/tfgen/generate_schema_test.go [hunk 2]
```diff
@@ -51,6 +51,15 @@ func TestRegressMiniRandom(t *testing.T) {
...
 }
 
+func TestNestedMaxItemsOne(t *testing.T) {
+	provider := testprovider.ProviderMiniCloudflare()
+	schema, err := GenerateSchema(provider, diag.DefaultSink(io.Discard, io.Discard, diag.FormatOptions{
+		Color: colors.Never,
+	}))
+	assert.NoError(t, err)
+	bridgetesting.AssertEqualsJSONFile(t, "test_data/regress-minicloudflare-schema.json", schema)
+}
```
Comment (iwahbe) [inline pkg/tfgen/generate_schema_test.go:61]: This behavior didn't change. I'm just codifying my understanding of the old behavior with a test.

File: pf/internal/schemashim/block_schema.go (hunk 1 above)
Comment (t0yv0) [reply pf/internal/schemashim/block_schema.go:87]: This is great. This is the right fix but I've simplified the function and added some targeted unit tests. There's an additional problem that set-nested blocks were still not working properly.  Consider: https://github.com/pulumi/pulumi-terraform-bridge/pull/1018

**Review by t0yv0 — APPROVED** [review]

File: pkg/tfshim/schema/schema.go [hunk 3]
```diff
@@ -168,3 +168,85 @@ func (m SchemaMap) Set(key string, value shim.Schema) {
...
+func FromProvider(p shim.Provider) *Provider {
+	if p == nil {
+		return nil
+	}
+	return &Provider{
+		Schema:         FromSchemaMap(p.Schema()),
+		ResourcesMap:   FromResourceMap(p.ResourcesMap()),
+		DataSourcesMap: FromResourceMap(p.DataSourcesMap()),
+	}
+}
```
Comment (iwahbe) [inline pkg/tfshim/schema/schema.go:252]: Before I merge, I don't understand where these functions are used. @t0yv0 Can you clarify?

File: pkg/tfshim/schema/schema.go [hunk 4]
```diff
@@ -168,3 +168,85 @@ func (m SchemaMap) Set(key string, value shim.Schema) {
 func (m SchemaMap) Delete(key string) {
 	delete(m, key)
 }
+
+func FromSchema(s shim.Schema) *Schema {
```
Comment (t0yv0) [inline pkg/tfshim/schema/schema.go:172]: This is not needed. Apologies. Should delete.

**Pull request merged**